File: jung-visualization/src/main/java/edu/uci/ics/jung/visualization/spatial/rtree/RTreeNode.java
package edu.uci.ics.jung.visualization.spatial.rtree;

import java.util.Optional;

/**
 * contains the parent for Node implementations
 *
 * @param <T>
 */
public abstract class RTreeNode<T> implements Node<T> {

  protected Optional<Node<T>> parent = Optional.empty();

  public void setParent(Node<T> node) {
    parent = Optional.of(node);
  }

  public Optional<Node<T>> getParent() {
    return parent;
  }
}

File: jung-visualization/src/main/kotlin/edu/uci/ics/jung/visualization/spatial/rtree/RTreeNode.kt
package edu.uci.ics.jung.visualization.spatial.rtree

import java.util.Optional

/**
 * contains the _parent for Node implementations
 *
 * @param T
 */
abstract class RTreeNode<T> : Node<T> {

  protected var _parent: Optional<Node<T>> = Optional.empty()

  override fun setParent(node: Node<T>) {
    _parent = Optional.of(node)
  }

  override fun getParent(): Optional<Node<T>> = _parent
}
